lib/hippie_csv/support.rb

```diff
@@ -28,16 +28,24 @@ def maybe_parse(string)
         end
       end
 
-      def maybe_stream(path, string)
+      def maybe_stream(path, &block)
+        sample_lines = File.foreach(path).first(10).join
```

Comment:
Read on stackoverflow that if you call `File.foreach` without a block, it only reads as much as it needs (i.e. the first 10 lines), and then closes the file.
See https://stackoverflow.com/a/24708453